core/src/test/scala/weaponregex/parser/ParserTest.scala

```diff
@@ -0,0 +1,20 @@
+package weaponregex.parser
+
+import weaponregex.model.regextree.RegexTree
+
+import scala.util.Failure
+
+abstract class ParserTest extends munit.FunSuite {
+  val parserFlavor: ParserFlavor
+
+  def treeBuildTest(tree: RegexTree, pattern: String): Unit = assertEquals(tree.build, pattern)
+
+  def parseErrorTest(pattern: String): Unit = {
+    val parsedTree = Parser(pattern, parserFlavor)
+
+    assert(clue(parsedTree) match {
+      case Failure(exception: RuntimeException) => exception.getMessage.startsWith("[Error] Parser:")
+      case _                                    => false
```

Comment:
If you match on the result and then do the assertion you can get clearer line reporting, and you can use a `fail("")` with a nicer message of what you are not expecting (instead of `assert(false)`

Reply:
Good to know. We will take that into account when we look at the tests.